criu/kerndat.c

```diff
@@ -1384,12 +1384,15 @@ static bool kerndat_has_clone3_set_tid(void)
 	pid_t pid;
 	struct _clone_args args = {};
 
+	kdat.has_clone3_set_tid = false;
+	if (opts.unprivileged && !has_cap_checkpoint_restore(opts.cap_eff))
```

Comment:
> Pretend it's not available instead of failing restore later when running unprivileged.

Can restore succeed if  kdat.has_clone3_set_tid isn't set and criu is running w/o cap_sys_admin and cap_cr?

Reply:
Yes. It does it using the old method: by using `ns_last_pid` (via a RPC helper).

Reply:
Let's write a comment to explain this in the code.

Reply:
I think this isn't a right solution for the problem:
* process can be restored in a separate user and pid namespaces and clone3() works fine in this case.
* this function checks whether clone3 is supported or not, and it is separate from the dump/restore code. I think we have to move these checks to the code that restores processes with specified pids.

Reply:
yes, you are right.